app/models/allocation.rb

```diff
@@ -50,22 +50,41 @@ class Allocation < VersionedModel
 
   attribute :complex_cases, Types::JSONB.new(Allocation::ComplexCaseAnswers)
 
-  def refresh_moves_count!
-    self.moves_count = moves.not_cancelled.count
-    save!
-  end
+  after_initialize :initialize_state
+  delegate :fill, :unfill, :cancel, to: :state_machine
 
-  def cancel
+  def cancel_with_moves
```

Comment:
by defining this, is there any danger with exposing `.cancel` and `.cancel_with_moves` ? in what circumstances should you call one over the other?

Reply:
That approach didn't work, I've amended as previously suggested: removing delegation from state machine and calling that explicitly instead